Automattic/jetpack
# VideoPress: Add support for new token bridge request id and token `ack` messages

Fixes # https://github.com/Automattic/greenhouse/issues/1375

#### Changes proposed in this Pull Request:

* Adds an acknowledgement post message event to the videopress token bridge in order to support cases where a videopress iframe loads and starts requesting tokens prior to the bridge being loaded (when the script is enqueued at the bottom of the page for example). The acknowledgement message allows the player to know if its request is being performed or not, which will allow it to decide if it needs to retry the request at a later time.

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


nope

#### Testing instructions:






* Test on a site with videopress
* rebuild the VideoPress package `jetpack build` -> `packages` -> `videopress`
* set the theme to `scratchpad` or any other theme that enqueues scrips to the bottom of the page
* ~~use the player built from https://github.com/Automattic/videopress/pull/809 (or prod if that PR has been deployed)~~ this change is now deployed to prod
* add a private video to a post or page
* view the public facing page with the video, the video should load and you should not see a `post message timed out` message
